com.microsoft.java.debug.core/src/main/java/com/microsoft/java/debug/core/adapter/handler/AttachRequestHandler.java

```diff
@@ -85,7 +86,9 @@
             options.put(Constants.PROJECTNAME, attachArguments.projectName);
         }
         ISourceLookUpProvider sourceProvider = context.getProvider(ISourceLookUpProvider.class);
-        sourceProvider.initialize(context.getDebugSession(), options);
+        sourceProvider.initialize(context, options);
```

Comment:
@testforstephen added a new handler that does post-processing after launching/attaching. Please consider putting the code there.

Reply:
Post launching is not implemented yet. That PR was closed.

Reply:
I think all providers should call initialize too

Reply:
Agree with @andxu, the context should have some interface to get all providers, then just iterate all providers and initialize them.

Reply:
The DebugAdapterContext should return a ProviderRegistry instead of a concrete getProvider method. Then invoke ProviderRegistry to iterate all registered providers.

Reply:
Just leave the IProvider/IProvdierContext refactor later and outside the scope of this change. We can have another PR for how to organize/initialize/uninstall IProviders.